app/models/agents/local_file_agent.rb

```diff
@@ -0,0 +1,186 @@
+module Agents
+  class LocalFileAgent < Agent
+    include LongRunnable
+    include FormConfigurable
+    include FileHandling
+
+    emits_file_pointer!
+
+    default_schedule 'every_1h'
+
+    def self.should_run?
+      ENV['ENABLE_INSECURE_AGENTS'] == "true"
+    end
+
+    description do
+      <<-MD
+        The LocalFileAgent can watch a file/directory for changes or emit an event for every file in that directory. When receiving an event it writes the received data into a file.
+
+        `mode` determines if the agent is emitting events for (changed) files or writing received event data to disk.
+
+        ### Reading
+
+        When `watch` is set to `true` the LocalFileAgent will watch the specified `path` for changes. An event will be emitted for every detected change.
```

Comment:
Watch is continuous, right, and so the check period doesn't matter in this case?  If so, maybe add:

"When in `watch` mode, the schedule is ignored and the file system is watched continuously."